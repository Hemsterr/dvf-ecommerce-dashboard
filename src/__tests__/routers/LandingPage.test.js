// Libs
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'

// Component
import LadingPage from '../../routers/landing-page'

// Constants
import keycodes from '../../constants/keyCodes'

const wrapper = (props = {}) => {
  const component = render(
    <Router>
      <LadingPage {...props} />
    </Router>
  )
  return component
}

describe('LadingPage component', () => {
  test('Render the LadingPage component with default props', () => {
    const wrapperWithoutProps = wrapper()
    expect(wrapperWithoutProps.container).toMatchSnapshot()
  })

  test('Testing event onBlur textinput', () => {
    const wrapperWithProps = wrapper()
    const { container, rerender } = wrapperWithProps

    // Checking event onBlur User name
    const formInput = container.querySelectorAll('input')
    expect(formInput.length).toEqual(2)

    const button = container.querySelector('.btn')
    fireEvent.click(button)
    rerender(
      <Router>
        <LadingPage />
      </Router>
    )
    const messageError = container.querySelectorAll('.input-group__label-error')
    expect(messageError.length).toEqual(2)
  })

  test('Testing render message error', () => {
    const wrapperWithProps = wrapper()
    const { container, rerender } = wrapperWithProps

    // Input form
    const formInput = container.querySelectorAll('input')
    const messageError = container.querySelectorAll('.input-group__label-error')
    const button = container.querySelector('.btn')
    fireEvent.blur(formInput[0], {
      target: { value: '0123456' },
    })

    fireEvent.click(button)

    // Handle event key down
    fireEvent.keyDown(formInput[1], {
      keyCode: keycodes.KEY_ENTER,
      target: { value: 'join@gmail.com' },
    })

    rerender(
      <Router>
        <LadingPage />
      </Router>
    )
    expect(messageError.length).toEqual(0)
  })
})
