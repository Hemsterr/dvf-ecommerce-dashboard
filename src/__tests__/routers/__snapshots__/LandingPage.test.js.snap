// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LadingPage component Render the LadingPage component with default props 1`] = `
<div>
  <div
    class="landing"
  >
    <div>
      <h1
        class="landing__hemster-logo"
      >
        <img
          alt="Hemster"
          src="[object Object]"
        />
      </h1>
      <p
        class="landing__header"
      >
        alterations
      </p>
      <h1
        class="landing__hemster-logo"
      >
        <img
          alt="DVF"
          src="[object Object]"
        />
      </h1>
      <p
        class="landing__title"
      >
        Please enter your order number:
      </p>
      <p
        class="landing__description"
      >
        (order # can be found in your confirmation email)
      </p>
      <div
        class="landing__input col-lg-4 col-xs-12 col-sm-12 input-group"
      >
        
        <div>
          <input
            class="input-group__text  input-group__without-border"
            placeholder=""
            type="text"
            value=""
          />
          
        </div>
      </div>
      <button
        class="btn btn__danger btn__medium"
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
